salt/modules/zfs.py

```diff
@@ -362,164 +293,229 @@ def list_(name=None, **kwargs):
-
-    # parse output
-    res = __salt__['cmd.run_all'](cmd)
+            opts['-S'] = kwargs.get('sort')
+    if isinstance(properties, list):
+        # NOTE: There can be only one -o and it takes a comma-seperated list
+        opts['-o'] = ','.join(properties)
+    else:
+        opts['-o'] = properties
+
+    ## parse zfs list
+    res = __salt__['cmd.run_all'](
+        __utils__['zfs.zfs_command'](
+            command='list',
+            flags=flags,
+            opts=opts,
+            target=name,
+        ),
+        python_shell=False,
+    )
     if res['retcode'] == 0:
-        for ds in [l for l in res['stdout'].splitlines()]:
-            ds = ds.split("\t")
-            ds_data = {}
-
-            for prop in properties:
-                ds_data[prop] = _conform_value(ds[properties.index(prop)])
+        for ds in res['stdout'].splitlines():
+            if kwargs.get('parsable', True):
+                ds_data = __utils__['zfs.from_auto_dict'](
+                    OrderedDict(list(zip(properties, ds.split("\t")))),
+                )
+            else:
+                ds_data = __utils__['zfs.to_auto_dict'](
+                    OrderedDict(list(zip(properties, ds.split("\t")))),
+                    convert_to_human=True,
+                )
 
             ret[ds_data['name']] = ds_data
             del ret[ds_data['name']]['name']
     else:
-        ret['error'] = res['stderr'] if 'stderr' in res else res['stdout']
+        return __utils__['zfs.parse_command_result'](res)
 
     return ret
 
 
-def mount(name='-a', **kwargs):
+def list_mount():
     '''
-    .. versionadded:: 2016.3.0
+    List mounted zfs filesystems
+
+    .. versionadded:: Fluorine
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' zfs.list_mount
 
+    '''
+    ## List mounted filesystem
+    res = __salt__['cmd.run_all'](
+        __utils__['zfs.zfs_command'](
+            command='mount',
+        ),
+        python_shell=False,
+    )
+
+    if res['retcode'] == 0:
+        ret = OrderedDict()
+        for mount in res['stdout'].splitlines():
+            mount = mount.split()
+            ret[mount[0]] = mount[-1]
+        return ret
+    else:
+        return __utils__['zfs.parse_command_result'](res)
+
+
+def mount(name=None, **kwargs):
+    '''
     Mounts ZFS file systems
 
     name : string
-        name of the filesystem, you can use '-a' to mount all unmounted filesystems. (this is the default)
+        name of the filesystem, having this set to None will mount all filesystems. (this is the default)
     overlay : boolean
         perform an overlay mount.
     options : string
         optional comma-separated list of mount options to use temporarily for
         the duration of the mount.
 
+    .. versionadded:: 2016.3.0
+    .. versionchanged:: Fluorine
+
+    .. warning::
+
+            Passing '-a' as name is deprecated and will be removed 2 verions after Flourine.
```

Comment:
Fixed, commit incoming. I also clarified the note om how the if statement should be updated when the time comes to avoid any confusion. I did not know about warn_until, this is handy.